src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -269,4 +275,90 @@ public CloudResource deleteControlledResource(String resourceName) {
     // TODO: change this method to call WSM controlled resource endpoints once they're ready
     return workspaceContext.listControlledResources();
   }
+
+  /**
+   * Lookup a data reference by its name. Names are unique within a workspace.
+   *
+   * @param referenceName name of reference to lookup
+   * @return the data reference object
+   * @throws RuntimeException if the resource is not a data reference (e.g. VM)
+   */
+  public CloudResource getDataReference(String referenceName) {
+    // TODO: change this method to call WSM data reference endpoints once they're ready
+    CloudResource dataReference = workspaceContext.getCloudResource(referenceName);
+    if (dataReference == null) {
+      throw new RuntimeException(referenceName + " not found.");
+    }
+    if (!dataReference.type.isDataReference) {
+      throw new RuntimeException(dataReference + " is not a data reference.");
+    }
+    return dataReference;
+  }
+
+  /**
+   * Add a new data reference in the workspace.
+   *
+   * <p>This method checks that the data reference exists using the user's credentials. The
+   * reference does not have to be within the workspace project.
+   *
+   * @param referenceType type of reference to add
+   * @param referenceName name of reference to add
+   * @return the data reference that was added
+   */
+  public CloudResource addDataReference(
+      CloudResource.Type referenceType, String referenceName, String cloudId) {
```

Comment:
maybe `cloudUri` instead of `cloudId`?

Reply:
I prefer cloudId here, since this is intended to handle other types of data references (e.g. BQ datasets) in the future. When I get to actually implementing those, maybe that will be awkward and it'll make more sense to split them up into different method, or maybe WSM endpoints will be ready and we'll call them. Either way, I prefer to keep it generic for now and specific later once we have more example resource types.